base/src/main/java/io/spine/io/Resource.java

```diff
@@ -48,20 +52,65 @@
 public final class Resource {
 
     private final String path;
+    private final @Nullable ClassLoader classLoader;
 
-    private Resource(String path) {
+    private Resource(String path, @Nullable ClassLoader classLoader) {
         this.path = path;
+        this.classLoader = classLoader;
     }
 
     /**
-     * Creates a new instance.
+     * Creates a new resource reference.
      *
      * @param path
-     *         the path to the resource file, relative to the classpath.
+     *         the path to the resource file, relative to the classpath
      */
     public static Resource file(String path) {
         checkNotEmptyOrBlank(path);
-        return new Resource(path);
+        return new Resource(path, null);
+    }
+
+    /**
+     * Creates a new reference to a resource at the context of the given class.
+     *
+     * @param path
+     *         the path to the resource file, relative to {@code contextClass}
+     * @param classLoader
+ *         the class relative to which the resource is referenced */
+    public static Resource file(String path, ClassLoader classLoader) {
+        checkNotNull(classLoader);
+        checkNotEmptyOrBlank(path);
+        return new Resource(path, classLoader);
+    }
+
+    private boolean inContext() {
+        return classLoader != null;
+    }
+
+    private @Nullable URL findUrl() {
+        URL url = classLoader().getResource(path);
+        return url;
+    }
+
+    private ClassLoader classLoader() {
+        return MoreObjects.firstNonNull(classLoader, Resource.class.getClassLoader());
+    }
+
+    private String classLoaderDiag() {
+        ClassLoader loader = classLoader();
+        String fmt = inContext() ? "the assigned class loader `%s`." : "the class loader `%s`.";
+        return format(fmt, loader);
+    }
+
+    private URL toUrl() {
+        URL url = findUrl();
+        if (url == null) {
```

Comment:
Let's just not allow `null` value. Instead, use `Resource.class.getClassLoader()` as the default. Please call `file(String, ClassLoader)` from `file(String)` with the default class loader. Then, we won't have to use `firstNonNull`.